app/services/call_event.rb

```diff
@@ -0,0 +1,47 @@
+module CallEvent
+  class Base
+    def self.publish(call)
+      new(call).publish
+    end
+
+    def initialize(call)
+      @call = call
+    end
+
+    def publish
+      ChampaignQueue.push(payload)
```

Comment:
It would be great if we could make this into a call to ActionBuilder to build an action with this payload and the hidden fields from the URL params.